flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/SortMergeSubpartitionReader.java

```diff
@@ -138,7 +138,9 @@ public void notifyDataAvailable() {
 
 	@Override
 	public void recycle(MemorySegment segment) {
-		readBuffers.add(segment);
+		if (!isReleased) {
+			readBuffers.add(segment);
+		}
```

Comment:
Do we need to return the segments to the `MemoryManager`? I guess not since they are unpooled.

Reply:
You are right, the buffers are unpooled so we do not need to return the segments to the ```MemoryManager```.